plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeAnalyze.java

```diff
@@ -695,6 +695,81 @@ public void testIncrementalStatisticsUpdateOnInsert()
+                "INSERT INTO " + tableName + " VALUES (11, 'aa')", 1);
+
+        assertUpdate("ANALYZE " + tableName);
+
+        assertQuery(
+                "SHOW STATS FOR " + tableName,
+                """
+                        VALUES
+                        ('col_int_1', null, 1.0, 0.0, null, 11, 11),
+                        ('col_varchar_1', 2.0, 1.0, 0.0, null, null, null),
+                        (null, null, null, null, 1.0, null, null)
+                        """);
+
+        assertUpdate("ALTER TABLE " + tableName + " ADD COLUMN col_int_2 bigint");
+        assertUpdate("ALTER TABLE " + tableName + " ADD COLUMN col_varchar_2 varchar");
+
+        assertUpdate(
+                disableStatisticsCollectionOnWrite(getSession()),
+                "INSERT INTO " + tableName + " VALUES (12, 'ab', 21, 'ba'), (13, 'ac', 22, 'bb')", 2);
+
+        assertUpdate("ANALYZE " + tableName);
+        assertQuery(
+                "SHOW STATS FOR " + tableName,
+                """
+                        VALUES
+                        ('col_int_1', null, 3.0, 0.0, null, 11, 13),
+                        ('col_varchar_1', 6.0, 3.0, 0.0, null, null, null),
+                        ('col_int_2', null, 2.0, 0.1, null, 21, 22),
+                        ('col_varchar_2', 4.0, 2.0, 0.1, null, null, null),
+                        (null, null, null, null, 3.0, null, null)
+                        """);
+    }
+
+    @Test
+    public void testStatsOnInsertWhenNewColumnAdded()
+    {
+        String tableName = "test_statistics_on_insert_when_new_column_added_" + randomNameSuffix();
+        assertUpdate("CREATE TABLE " + tableName + " (col_int_1 bigint, col_varchar_1 varchar)");
+
+        assertUpdate("INSERT INTO " + tableName + " VALUES (11, 'aa')", 1);
+
+        assertQuery(
+                "SHOW STATS FOR " + tableName,
+                """
+                        VALUES
+                        ('col_int_1', null, 1.0, 0.0, null, 11, 11),
+                        ('col_varchar_1', 2.0, 1.0, 0.0, null, null, null),
+                        (null, null, null, null, 1.0, null, null)
+                        """);
+
+        assertUpdate("ALTER TABLE " + tableName + " ADD COLUMN col_int_2 bigint");
+        assertUpdate("ALTER TABLE " + tableName + " ADD COLUMN col_varchar_2 varchar");
+
+        assertUpdate("INSERT INTO " + tableName + " VALUES (12, 'ab', 21, 'ba'), (13, 'ac', 22, 'bb')", 2);
+
+        assertQuery(
+                "SHOW STATS FOR " + tableName,
+                """
+                        VALUES
+                        ('col_int_1', null, 3.0, 0.0, null, 11, 13),
+                        ('col_varchar_1', 6.0, 3.0, 0.0, null, null, null),
+                        ('col_int_2', null, 2.0, 0.1, null, 21, 22),
+                        ('col_varchar_2', 4.0, 2.0, 0.1, null, null, null),
+                        (null, null, null, null, 3.0, null, null)
+                        """);
```

Comment:
Would be nice to convey that stats are same, regardless whether we collect on write, or ANALYZE.

Perhaps the test could be structured as follows

```java
@Test(dataProviderClass = DataProviders.class, dataProvider = "trueFalse")
    public void testCollectStatsAfterColumnAdded(boolean collectOnWrite)
    {
        Session session = getSession();
        String tableName = "test_collect_stats_after_column_added" + randomNameSuffix();
        assertUpdate("CREATE TABLE " + tableName + " (col_int_1 bigint, col_varchar_1 varchar)");

        assertUpdate("INSERT INTO " + tableName + " VALUES (11, 'aa')", 1);

        // Verify stats implicitly collected
        assertQuery(
                "SHOW STATS FOR " + tableName,
                """
                        VALUES
                        ('col_int_1', null, 1.0, 0.0, null, 11, 11),
                        ('col_varchar_1', 2.0, 1.0, 0.0, null, null, null),
                        (null, null, null, null, 1.0, null, null)
                        """);

        assertUpdate("ALTER TABLE " + tableName + " ADD COLUMN col_int_2 bigint");
        assertUpdate("ALTER TABLE " + tableName + " ADD COLUMN col_varchar_2 varchar");

        assertUpdate(
                Session.builder(session)
                        .setCatalogSessionProperty(session.getCatalog().orElseThrow(), EXTENDED_STATISTICS_COLLECT_ON_WRITE, Boolean.toString(collectOnWrite))
                        .build(),
                "INSERT INTO " + tableName + " VALUES (12, 'ab', 21, 'ba'), (13, 'ac', 22, 'bb')",
                2);

        if (!collectOnWrite) {
            assertQuery(
                    "SHOW STATS FOR " + tableName,
                    """
                            VALUES
                            ('col_int_1', null, 1.0, 0.0, null, 11, 13),
                            ('col_varchar_1', 2.0, 1.0, 0.0, null, null, null),
                            ('col_int_2', null, null, null, null, 21, 22),
                            ('col_varchar_2', null, null, null, null, null, null),
                            (null, null, null, null, 3.0, null, null)
                            """);

            assertUpdate("ANALYZE " + tableName);
        }

        assertQuery(
                "SHOW STATS FOR " + tableName,
                """
                        VALUES
                        ('col_int_1', null, 3.0, 0.0, null, 11, 13),
                        ('col_varchar_1', 6.0, 3.0, 0.0, null, null, null),
                        ('col_int_2', null, 2.0, 0.1, null, 21, 22),
                        ('col_varchar_2', 4.0, 2.0, 0.1, null, null, null),
                        (null, null, null, null, 3.0, null, null)
                        """);
    }
```